File: agents/linux/agent.go
package main

import (
    "bytes"
    "encoding/json"
    "fmt"
    "io/ioutil"
    "log"
    "net/http"
    "os"
    "os/signal"
    "runtime"
    "syscall"
    "time"

    "github.com/shirou/gopsutil/v3/cpu"
    "github.com/shirou/gopsutil/v3/disk"
    "github.com/shirou/gopsutil/v3/host"
    "github.com/shirou/gopsutil/v3/mem"
    "github.com/shirou/gopsutil/v3/net"
    "github.com/shirou/gopsutil/v3/process"
    "gopkg.in/yaml.v3"
)

// ==================== CONFIGURATION ====================

type Config struct {
    ServerID   string `yaml:"server_id"`
    MainServer string `yaml:"main_server"`
    APIKey     string `yaml:"api_key"`
    Interval   int    `yaml:"interval"`
    LogLevel   string `yaml:"log_level"`
    TLS        struct {
        Enabled  bool   `yaml:"enabled"`
        CertFile string `yaml:"cert_file"`
        KeyFile  string `yaml:"key_file"`
    } `yaml:"tls"`
}

type Agent struct {
    Config     Config
    HTTPClient *http.Client
    StopChan   chan struct{}
}

// ==================== METRICS STRUCTURES ====================

type Metrics struct {
    ServerID      string                 `json:"server_id"`
    Timestamp     int64                  `json:"timestamp"`
    CPU           float64                `json:"cpu"`
    Memory        float64                `json:"memory"`
    Disk          float64                `json:"disk"`
    LoadAvg       []float64              `json:"load_avg"`
    ProcessCount  int32                  `json:"process_count"`
    Network       []NetworkStats         `json:"network"`
    OS            string                 `json:"os"`
    OSVersion     string                 `json:"os_version"`
    Hostname      string                 `json:"hostname"`
    Uptime        uint64                 `json:"uptime"`
    CustomMetrics map[string]interface{} `json:"custom_metrics"`
}

type NetworkStats struct {
    Name        string `json:"name"`
    BytesSent   uint64 `json:"bytes_sent"`
    BytesRecv   uint64 `json:"bytes_recv"`
    PacketsSent uint64 `json:"packets_sent"`
    PacketsRecv uint64 `json:"packets_recv"`
}

type RegisterData struct {
    ServerID  string `json:"server_id"`
    Hostname  string `json:"hostname"`
    OS        string `json:"os"`
    OSVersion string `json:"os_version"`
    IP        string `json:"ip"`
    Version   string `json:"version"`
}

// ==================== MAIN FUNCTIONS ====================

func main() {
    // Load configuration
    configPath := os.Getenv("CONFIG_PATH")
    if configPath == "" {
        configPath = "/etc/nexora/agent.yaml"
    }

    config, err := loadConfig(configPath)
    if err != nil {
        log.Printf("❌ Failed to load config: %v", err)
        log.Println("Using default configuration...")
        config = getDefaultConfig()
    }

    // Create agent
    agent := NewAgent(config)

    // Handle signals
    signalChan := make(chan os.Signal, 1)
    signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)

    go func() {
        <-signalChan
        log.Println("📴 Received shutdown signal, stopping agent...")
        agent.Stop()
    }()

    // Start agent
    log.Printf("🚀 Nexora Agent Starting...")
    log.Printf("📡 Server ID: %s", config.ServerID)
    log.Printf("🌐 Main Server: %s", config.MainServer)
    log.Printf("⏱️  Interval: %d seconds", config.Interval)
    log.Printf("💻 OS: %s", runtime.GOOS)
    log.Println("─────────────────────────────────────────")

    agent.Start()
}

// ==================== AGENT METHODS ====================

func NewAgent(config Config) *Agent {
    return &Agent{
        Config: config,
        HTTPClient: &http.Client{
            Timeout: 10 * time.Second,
        },
        StopChan: make(chan struct{}),
    }
}

func (a *Agent) Start() {
    // Register agent with server
    if err := a.Register(); err != nil {
        log.Printf("⚠️ Registration failed: %v", err)
    }

    // Start metrics collection
    ticker := time.NewTicker(time.Duration(a.Config.Interval) * time.Second)
    defer ticker.Stop()

    // Send first metrics immediately
    a.CollectAndSend()

    for {
        select {
        case <-ticker.C:
            a.CollectAndSend()
        case <-a.StopChan:
            return
        }
    }
}

func (a *Agent) Stop() {
    close(a.StopChan)
    log.Println("🛑 Agent stopped")
}

func (a *Agent) Register() error {
    hostname, _ := os.Hostname()
    ip := getLocalIP()

    registerData := RegisterData{
        ServerID:  a.Config.ServerID,
        Hostname:  hostname,
        OS:        runtime.GOOS,
        OSVersion: getOSVersion(),
        IP:        ip,
        Version:   "1.0.0",
    }

    jsonData, err := json.Marshal(registerData)
    if err != nil {
        return err
    }

    req, err := http.NewRequest("POST", 
        fmt.Sprintf("%s/api/agents/linux/register", a.Config.MainServer), 
        bytes.NewBuffer(jsonData))
    if err != nil {
        return err
    }

    req.Header.Set("Content-Type", "application/json")
    req.Header.Set("X-API-Key", a.Config.APIKey)
    req.Header.Set("X-Server-ID", a.Config.ServerID)

    resp, err := a.HTTPClient.Do(req)
    if err != nil {
        return err
    }
    defer resp.Body.Close()

    if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusCreated {
        body, _ := ioutil.ReadAll(resp.Body)
        return fmt.Errorf("registration failed: %d - %s", resp.StatusCode, string(body))
    }

    log.Printf("✅ Agent registered successfully with server: %s", a.Config.MainServer)
    return nil
}

func (a *Agent) CollectAndSend() {
    metrics, err := a.CollectMetrics()
    if err != nil {
        log.Printf("❌ Failed to collect metrics: %v", err)
        return
    }

    if err := a.SendMetrics(metrics); err != nil {
        log.Printf("❌ Failed to send metrics: %v", err)
        return
    }

    log.Printf("✅ Sent metrics: CPU=%.1f%% MEM=%.1f%% DISK=%.1f%%",
        metrics.CPU, metrics.Memory, metrics.Disk)
}

func (a *Agent) CollectMetrics() (*Metrics, error) {
    // CPU
    cpuPercent, err := cpu.Percent(time.Second, false)
    if err != nil {
        return nil, err
    }

    // Memory
    memInfo, err := mem.VirtualMemory()
    if err != nil {
        return nil, err
    }

    // Disk
    diskInfo, err := disk.Usage("/")
    if err != nil {
        return nil, err
    }

    // Load Average
    loadAvg, err := cpu.LoadAvg()
    if err != nil {
        loadAvg = &cpu.LoadAvgStat{}
    }

    // Process Count
    processes, err := process.Processes()
    if err != nil {
        return nil, err
    }

    // Network Stats
    netStats, err := net.IOCounters(true)
    if err != nil {
        return nil, err
    }

    var networks []NetworkStats
    for _, stat := range netStats {
        if stat.BytesSent > 0 || stat.BytesRecv > 0 {
            networks = append(networks, NetworkStats{
                Name:        stat.Name,
                BytesSent:   stat.BytesSent,
                BytesRecv:   stat.BytesRecv,
                PacketsSent: stat.PacketsSent,
                PacketsRecv: stat.PacketsRecv,
            })
        }
    }

    // Host Info
    hostInfo, _ := host.Info()
    hostname, _ := os.Hostname()

    // Custom metrics
    customMetrics := make(map[string]interface{})
    if runtime.GOOS == "linux" {
        customMetrics["kernel"] = hostInfo.KernelVersion
        customMetrics["platform"] = hostInfo.Platform
        customMetrics["platform_version"] = hostInfo.PlatformVersion
        customMetrics["cpu_count"] = runtime.NumCPU()
    }

    return &Metrics{
        ServerID:      a.Config.ServerID,
        Timestamp:     time.Now().Unix(),
        CPU:           cpuPercent[0],
        Memory:        memInfo.UsedPercent,
        Disk:          diskInfo.UsedPercent,
        LoadAvg:       []float64{loadAvg.Load1, loadAvg.Load5, loadAvg.Load15},
        ProcessCount:  int32(len(processes)),
        Network:       networks,
        OS:            runtime.GOOS,
        OSVersion:     hostInfo.PlatformVersion,
        Hostname:      hostname,
        Uptime:        hostInfo.Uptime,
        CustomMetrics: customMetrics,
    }, nil
}

func (a *Agent) SendMetrics(metrics *Metrics) error {
    jsonData, err := json.Marshal(metrics)
    if err != nil {
        return err
    }

    req, err := http.NewRequest("POST", 
        fmt.Sprintf("%s/api/agents/linux/metrics", a.Config.MainServer), 
        bytes.NewBuffer(jsonData))
    if err != nil {
        return err
    }

    req.Header.Set("Content-Type", "application/json")
    req.Header.Set("X-API-Key", a.Config.APIKey)
    req.Header.Set("X-Server-ID", a.Config.ServerID)

    resp, err := a.HTTPClient.Do(req)
    if err != nil {
        return err
    }
    defer resp.Body.Close()

    if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusCreated {
        body, _ := ioutil.ReadAll(resp.Body)
        return fmt.Errorf("server returned: %d - %s", resp.StatusCode, string(body))
    }

    return nil
}

// ==================== HELPER FUNCTIONS ====================

func loadConfig(path string) (Config, error) {
    data, err := ioutil.ReadFile(path)
    if err != nil {
        return Config{}, err
    }

    var config Config
    if err := yaml.Unmarshal(data, &config); err != nil {
        return Config{}, err
    }

    return config, nil
}

func getDefaultConfig() Config {
    serverID := os.Getenv("SERVER_ID")
    if serverID == "" {
        hostname, _ := os.Hostname()
        serverID = hostname
    }

    mainServer := os.Getenv("MAIN_SERVER")
    if mainServer == "" {
        mainServer = "https://nexora.yourdomain.com"
    }

    apiKey := os.Getenv("API_KEY")
    interval := 30

    return Config{
        ServerID:   serverID,
        MainServer: mainServer,
        APIKey:     apiKey,
        Interval:   interval,
        LogLevel:   "info",
    }
}

func getLocalIP() string {
    addrs, err := net.Interfaces()
    if err != nil {
        return "unknown"
    }

    for _, addr := range addrs {
        if addr.Flags&net.FlagUp != 0 && addr.Flags&net.FlagLoopback == 0 {
            if ip, ok := addr.Addrs[0].(*net.IPNet); ok && ip.IP.To4() != nil {
                return ip.IP.String()
            }
        }
    }
    return "unknown"
}

func getOSVersion() string {
    hostInfo, err := host.Info()
    if err != nil {
        return runtime.GOOS
    }
    return hostInfo.PlatformVersion
}

File: agents/windows/agent.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Management;
using System.Net.Http;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace NexoraAgent
{
    // ==================== CONFIGURATION ====================

    public class Config
    {
        public string ServerId { get; set; }
        public string MainServer { get; set; }
        public string ApiKey { get; set; }
        public int Interval { get; set; }
        public string LogLevel { get; set; }
        public bool UseTLS { get; set; }
    }

    // ==================== METRICS STRUCTURES ====================

    public class Metrics
    {
        [JsonProperty("server_id")]
        public string ServerId { get; set; }

        [JsonProperty("timestamp")]
        public long Timestamp { get; set; }

        [JsonProperty("cpu")]
        public double Cpu { get; set; }

        [JsonProperty("memory")]
        public double Memory { get; set; }

        [JsonProperty("disk")]
        public double Disk { get; set; }

        [JsonProperty("load_avg")]
        public double[] LoadAvg { get; set; }

        [JsonProperty("process_count")]
        public int ProcessCount { get; set; }

        [JsonProperty("network")]
        public List<NetworkStats> Network { get; set; }

        [JsonProperty("os")]
        public string OS { get; set; }

        [JsonProperty("os_version")]
        public string OSVersion { get; set; }

        [JsonProperty("hostname")]
        public string Hostname { get; set; }

        [JsonProperty("uptime")]
        public long Uptime { get; set; }

        [JsonProperty("custom_metrics")]
        public Dictionary<string, object> CustomMetrics { get; set; }
    }

    public class NetworkStats
    {
        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("bytes_sent")]
        public long BytesSent { get; set; }

        [JsonProperty("bytes_recv")]
        public long BytesRecv { get; set; }

        [JsonProperty("packets_sent")]
        public long PacketsSent { get; set; }

        [JsonProperty("packets_recv")]
        public long PacketsRecv { get; set; }
    }

    public class RegisterData
    {
        [JsonProperty("server_id")]
        public string ServerId { get; set; }

        [JsonProperty("hostname")]
        public string Hostname { get; set; }

        [JsonProperty("os")]
        public string OS { get; set; }

        [JsonProperty("os_version")]
        public string OSVersion { get; set; }

        [JsonProperty("ip")]
        public string IP { get; set; }

        [JsonProperty("version")]
        public string Version { get; set; }
    }

    // ==================== MAIN AGENT CLASS ====================

    public class Agent
    {
        private readonly Config _config;
        private readonly HttpClient _httpClient;
        private readonly CancellationTokenSource _cancellationTokenSource;
        private bool _isRunning;

        public Agent(Config config)
        {
            _config = config;
            _httpClient = new HttpClient
            {
                Timeout = TimeSpan.FromSeconds(10)
            };
            _cancellationTokenSource = new CancellationTokenSource();
            _isRunning = false;
        }

        public async Task StartAsync()
        {
            Console.WriteLine("🚀 Nexora Agent Starting...");
            Console.WriteLine($"📡 Server ID: {_config.ServerId}");
            Console.WriteLine($"🌐 Main Server: {_config.MainServer}");
            Console.WriteLine($"⏱️  Interval: {_config.Interval} seconds");
            Console.WriteLine($"💻 OS: Windows");
            Console.WriteLine("─────────────────────────────────────────");

            _isRunning = true;

            // Register agent
            await RegisterAsync();

            // Start metrics collection
            await CollectAndSendLoop();
        }

        public void Stop()
        {
            Console.WriteLine("🛑 Stopping agent...");
            _isRunning = false;
            _cancellationTokenSource.Cancel();
        }

        private async Task CollectAndSendLoop()
        {
            // Send first metrics immediately
            await CollectAndSendAsync();

            while (_isRunning && !_cancellationTokenSource.Token.IsCancellationRequested)
            {
                try
                {
                    await Task.Delay(_config.Interval * 1000, _cancellationTokenSource.Token);
                    await CollectAndSendAsync();
                }
                catch (TaskCanceledException)
                {
                    break;
                }
                catch (Exception ex)
                {
                    Console.WriteLine($"❌ Error in collection loop: {ex.Message}");
                }
            }
        }

        public async Task RegisterAsync()
        {
            try
            {
                var registerData = new RegisterData
                {
                    ServerId = _config.ServerId,
                    Hostname = Environment.MachineName,
                    OS = "Windows",
                    OSVersion = Environment.OSVersion.VersionString,
                    IP = GetLocalIPAddress(),
                    Version = "1.0.0"
                };

                var json = JsonConvert.SerializeObject(registerData);
                var content = new StringContent(json, Encoding.UTF8, "application/json");

                var request = new HttpRequestMessage(HttpMethod.Post, 
                    $"{_config.MainServer}/api/agents/windows/register");
                request.Headers.Add("X-API-Key", _config.ApiKey);
                request.Headers.Add("X-Server-ID", _config.ServerId);
                request.Content = content;

                var response = await _httpClient.SendAsync(request);
                if (response.IsSuccessStatusCode)
                {
                    Console.WriteLine($"✅ Agent registered successfully with server: {_config.MainServer}");
                }
                else
                {
                    var error = await response.Content.ReadAsStringAsync();
                    Console.WriteLine($"⚠️ Registration failed: {response.StatusCode} - {error}");
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine($"⚠️ Registration error: {ex.Message}");
            }
        }

        private async Task CollectAndSendAsync()
        {
            try
            {
                var metrics = CollectMetrics();
                await SendMetricsAsync(metrics);
                Console.WriteLine($"✅ Sent metrics: CPU={metrics.Cpu:F1}% MEM={metrics.Memory:F1}% DISK={metrics.Disk:F1}%");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"❌ Failed to collect/send metrics: {ex.Message}");
            }
        }

        private Metrics CollectMetrics()
        {
            var metrics = new Metrics
            {
                ServerId = _config.ServerId,
                Timestamp = DateTimeOffset.UtcNow.ToUnixTimeSeconds(),
                OS = "Windows",
                OSVersion = Environment.OSVersion.VersionString,
                Hostname = Environment.MachineName,
                CustomMetrics = new Dictionary<string, object>(),
                Network = new List<NetworkStats>()
            };

            // CPU Usage
            try
            {
                using (var cpuCounter = new PerformanceCounter("Processor", "% Processor Time", "_Total"))
                {
                    cpuCounter.NextValue(); // First call returns 0
                    Thread.Sleep(100);
                    metrics.Cpu = Math.Round(cpuCounter.NextValue(), 2);
                }
            }
            catch
            {
                metrics.Cpu = 0;
            }

            // Memory Usage
            try
            {
                using (var searcher = new ManagementObjectSearcher("SELECT TotalVisibleMemorySize, FreePhysicalMemory FROM Win32_ComputerSystem"))
                {
                    var result = searcher.Get().GetEnumerator();
                    if (result.MoveNext())
                    {
                        var total = Convert.ToDouble(result.Current["TotalVisibleMemorySize"]);
                        var free = Convert.ToDouble(result.Current["FreePhysicalMemory"]);
                        metrics.Memory = Math.Round((total - free) / total * 100, 2);
                    }
                }
            }
            catch
            {
                metrics.Memory = 0;
            }

            // Disk Usage
            try
            {
                using (var searcher = new ManagementObjectSearcher("SELECT Size, FreeSpace FROM Win32_LogicalDisk WHERE DriveType=3 AND DeviceID='C:'"))
                {
                    var result = searcher.Get().GetEnumerator();
                    if (result.MoveNext())
                    {
                        var total = Convert.ToDouble(result.Current["Size"]);
                        var free = Convert.ToDouble(result.Current["FreeSpace"]);
                        metrics.Disk = Math.Round((total - free) / total * 100, 2);
                    }
                }
            }
            catch
            {
                metrics.Disk = 0;
            }

            // Load Average (Windows doesn't have traditional load avg)
            metrics.LoadAvg = new double[] { 0, 0, 0 };

            // Process Count
            try
            {
                metrics.ProcessCount = Process.GetProcesses().Length;
            }
            catch
            {
                metrics.ProcessCount = 0;
            }

            // Network Stats
            try
            {
                using (var searcher = new ManagementObjectSearcher("SELECT Name, BytesSentPerSec, BytesReceivedPerSec FROM Win32_PerfFormattedData_Tcpip_NetworkInterface"))
                {
                    var results = searcher.Get();
                    foreach (ManagementObject obj in results)
                    {
                        var name = obj["Name"]?.ToString() ?? "Unknown";
                        var bytesSent = Convert.ToInt64(obj["BytesSentPerSec"] ?? 0);
                        var bytesRecv = Convert.ToInt64(obj["BytesReceivedPerSec"] ?? 0);
                        
                        if (bytesSent > 0 || bytesRecv > 0)
                        {
                            metrics.Network.Add(new NetworkStats
                            {
                                Name = name,
                                BytesSent = bytesSent,
                                BytesRecv = bytesRecv,
                                PacketsSent = 0,
                                PacketsRecv = 0
                            });
                        }
                    }
                }
            }
            catch
            {
                // Network stats unavailable
            }

            // Uptime
            try
            {
                using (var searcher = new ManagementObjectSearcher("SELECT LastBootUpTime FROM Win32_OperatingSystem"))
                {
                    var result = searcher.Get().GetEnumerator();
                    if (result.MoveNext())
                    {
                        var bootTime = ManagementDateTimeConverter.ToDateTime(result.Current["LastBootUpTime"].ToString());
                        metrics.Uptime = (long)(DateTime.Now - bootTime).TotalSeconds;
                    }
                }
            }
            catch
            {
                metrics.Uptime = 0;
            }

            // Windows Custom Metrics
            metrics.CustomMetrics["windows_version"] = Environment.OSVersion.Version.ToString();
            metrics.CustomMetrics["architecture"] = Environment.Is64BitOperatingSystem ? "64-bit" : "32-bit";
            metrics.CustomMetrics["processor_count"] = Environment.ProcessorCount;

            return metrics;
        }

        private async Task SendMetricsAsync(Metrics metrics)
        {
            var json = JsonConvert.SerializeObject(metrics);
            var content = new StringContent(json, Encoding.UTF8, "application/json");

            var request = new HttpRequestMessage(HttpMethod.Post, 
                $"{_config.MainServer}/api/agents/windows/metrics");
            request.Headers.Add("X-API-Key", _config.ApiKey);
            request.Headers.Add("X-Server-ID", _config.ServerId);
            request.Content = content;

            var response = await _httpClient.SendAsync(request);
            if (!response.IsSuccessStatusCode)
            {
                var error = await response.Content.ReadAsStringAsync();
                throw new Exception($"Server returned {response.StatusCode}: {error}");
            }
        }

        private string GetLocalIPAddress()
        {
            try
            {
                using (var searcher = new ManagementObjectSearcher("SELECT IPAddress FROM Win32_NetworkAdapterConfiguration WHERE IPEnabled=True"))
                {
                    var results = searcher.Get();
                    foreach (ManagementObject obj in results)
                    {
                        var addresses = obj["IPAddress"] as string[];
                        if (addresses != null)
                        {
                            foreach (var ip in addresses)
                            {
                                if (!ip.StartsWith("169.254") && !ip.StartsWith("::"))
                                {
                                    return ip;
                                }
                            }
                        }
                    }
                }
            }
            catch
            {
                // Return default
            }
            return "unknown";
        }
    }

    // ==================== PROGRAM ENTRY POINT ====================

    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("🚀 Nexora Windows Agent");
            Console.WriteLine("=========================");

            try
            {
                // Load configuration
                var config = LoadConfig();

                // Create and start agent
                var agent = new Agent(config);

                // Handle Ctrl+C
                Console.CancelKeyPress += (sender, e) =>
                {
                    Console.WriteLine("📴 Received shutdown signal...");
                    agent.Stop();
                    e.Cancel = true;
                };

                // Run agent
                agent.StartAsync().Wait();
            }
            catch (Exception ex)
            {
                Console.WriteLine($"❌ Fatal error: {ex.Message}");
                Console.WriteLine(ex.StackTrace);
                Environment.ExitCode = 1;
            }
        }

        static Config LoadConfig()
        {
            var config = new Config
            {
                ServerId = Environment.GetEnvironmentVariable("SERVER_ID") ?? Environment.MachineName,
                MainServer = Environment.GetEnvironmentVariable("MAIN_SERVER") ?? "https://nexora.yourdomain.com",
                ApiKey = Environment.GetEnvironmentVariable("API_KEY") ?? "",
                Interval = int.TryParse(Environment.GetEnvironmentVariable("INTERVAL"), out var interval) ? interval : 30,
                LogLevel = Environment.GetEnvironmentVariable("LOG_LEVEL") ?? "info",
                UseTLS = true
            };

            // Try to load from config file
            var configPath = Path.Combine(
                Environment.GetFolderPath(Environment.SpecialFolder.ProgramData),
                "Nexora", "agent.json");

            if (File.Exists(configPath))
            {
                try
                {
                    var json = File.ReadAllText(configPath);
                    var fileConfig = JsonConvert.DeserializeObject<Config>(json);
                    if (fileConfig != null)
                    {
                        // Merge configurations
                        if (!string.IsNullOrEmpty(fileConfig.ServerId)) config.ServerId = fileConfig.ServerId;
                        if (!string.IsNullOrEmpty(fileConfig.MainServer)) config.MainServer = fileConfig.MainServer;
                        if (!string.IsNullOrEmpty(fileConfig.ApiKey)) config.ApiKey = fileConfig.ApiKey;
                        if (fileConfig.Interval > 0) config.Interval = fileConfig.Interval;
                        if (!string.IsNullOrEmpty(fileConfig.LogLevel)) config.LogLevel = fileConfig.LogLevel;
                    }
                }
                catch (Exception ex)
                {
                    Console.WriteLine($"⚠️ Error loading config file: {ex.Message}");
                }
            }

            if (string.IsNullOrEmpty(config.ApiKey))
            {
                Console.WriteLine("❌ API Key is required");
                Console.WriteLine("Set API_KEY environment variable or create config file");
                Environment.Exit(1);
            }

            return config;
        }
    }
}